src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -115,10 +115,10 @@ public virtual TValue[] Get(TKey[] keys, Func<TKey, TValue> factory)
                     for (var i = 0; i < items.Length; i++)
                     {
                         TKey key = keys[i];
-                        TValue value = items[i];
-                        if (value == null)
+                        TValue value = default(TValue);
```

Comment:
Why not `null` explicitly?

Reply:
The TValue generic type doesn’t have the constraint of nullable/non-nullable. So the default value is the recommended way to go.

Reply:
On second thought, in that case, we should check for 
`if (value == default(TValue)` before adding into `fetched` collection

Reply:
Noted; I see that it is consistent with `TValue item = default(TValue)` in the `Get` single key method.

Also, it is appropriate to rename `value` to `item` here ⁠— a bit more diff in exchange for consistency.

Reply:
Oh wait, the default value of `item` in the `Get` single key method is not used.

> On second thought, in that case, we should probably check for
> if (value == default(TValue) before adding into fetched collection

Assigning to `null` is more straightforward, since `default(TValue)` can be a cachable value.

Reply:
> Assigning to null is more straightforward, since default(TValue) can be a cachable value.

`TValue value = null;`  -> this won't be able to compile as roslyn can't determine if TValue is nullable or non-nullable type.

I tried creating a test case on TValue => int/int?, and as expected, comparing `TValue value == null` doesn't work.

I had to use `EqualityComparer<TValue>.Default.Equals(value, default)` to compare if the generic type object == default value